src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/HaFlowPingSpec.groovy

```diff
@@ -0,0 +1,120 @@
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.extension.failfast.Tidy
+import org.openkilda.functionaltests.extension.tags.Tags
+import org.openkilda.functionaltests.helpers.HaFlowHelper
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.model.SwitchId
+import org.openkilda.model.cookie.Cookie
+import org.openkilda.northbound.dto.v2.haflows.HaFlowPingPayload
+import org.openkilda.northbound.dto.v2.haflows.HaFlowPingResult
+import org.openkilda.northbound.dto.v2.yflows.SubFlowPingPayload
+import org.openkilda.northbound.dto.v2.yflows.UniSubFlowPingPayload
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import spock.lang.Narrative
+import spock.lang.Shared
+import spock.lang.Ignore
+
+@Narrative("""This spec tests 'periodic ping' functionality.""")
+class HaFlowPingSpec extends HealthCheckSpecification {
+    @Autowired
+    @Shared
+    HaFlowHelper haFlowHelper
+
+    @Value('${flow.ping.interval}')
+    int pingInterval
+
+    @Tidy
+    @Ignore("https://github.com/telstra/open-kilda/issues/5217")
+    def "Able to turn on periodic pings on a ha-flow"() {
+        when: "Create a ha-flow with periodic pings turned on"
+        def swT = topologyHelper.switchTriplets.first()
+        def haFlowRequest = haFlowHelper.randomHaFlow(swT).tap {
+            it.periodicPings = true
+        }
+        def haFlow = haFlowHelper.addHaFlow(haFlowRequest)
+
+        then: "Periodic pings is really enabled"
+        northboundV2.getHaFlow(haFlow.haFlowId).periodicPings
+
+        and: "Packet counter on catch ping rules grows due to pings happening"
+        def sharedSwitchPacketCount = getPacketCountOfVlanPingRule(swT.shared.dpId)
+        def ep1SwitchPacketCount = getPacketCountOfVlanPingRule(swT.ep1.dpId)
+        def ep2SwitchPacketCount = getPacketCountOfVlanPingRule(swT.ep2.dpId)
+
+        Wrappers.wait(pingInterval + WAIT_OFFSET) {
+            def sharedPacketCountNow = getPacketCountOfVlanPingRule(swT.shared.dpId)
+            def ep1PacketCountNow = getPacketCountOfVlanPingRule(swT.ep1.dpId)
+            def ep2PacketCountNow = getPacketCountOfVlanPingRule(swT.ep2.dpId)
+
+            sharedPacketCountNow > sharedSwitchPacketCount && ep1PacketCountNow > ep1SwitchPacketCount &&
+                    ep2PacketCountNow > ep2SwitchPacketCount
+        }
+
+        cleanup:
+        haFlow && haFlowHelper.deleteHaFlow(haFlow.haFlowId)
+    }
+
+    @Tidy
+    @Ignore("unignore after https://github.com/telstra/open-kilda/issues/5224")
+    @Tags([LOW_PRIORITY])
+    def "Able to ping ha-flow when one of subflows is one-switch one"() {
+        given: "ha-flow which has one-switch subflow"
+        def switchTriplet = topologyHelper.getSwitchTriplets(true, true).find {
+            it.shared == it.ep1 && it.shared != it.ep2
```

Comment:
Move this filter to SwitchTriplet class.

Reply:
Please to it in separate refactoring PR

Reply:
I'm not sure if we should leave refactoring 'for later'. You can open a new PR with fixes of application's code from this one, and me or Artsiom polish these tests.